assignment/impl/src/java/org/sakaiproject/assignment/impl/GradeSheetExporter.java

```diff
@@ -70,7 +74,28 @@
     @Setter private FormattedText formattedText;
 
     private ResourceLoader rb = new ResourceLoader("assignment");
+    
+    /**
+    * A comparator that sorts by student sortName
+    */
+    static final Comparator<Submitter> SUBMITTER_NAME_COMPARATOR = new Comparator<Submitter>() {
+        Collator collator;
+        {
+            this.collator = Collator.getInstance();
+            try {
+                this.collator = new RuleBasedCollator(
+                        ((RuleBasedCollator) this.collator).getRules().replaceAll("<'\u005f'", "<' '<'\u005f'"));
+            } catch (final ParseException e) {
+                log.warn(this + " Cannot init RuleBasedCollator. Will use the default Collator instead.", e);
+            }
+        }
 
+        @Override
+        public int compare(final Submitter s1, final Submitter s2) {
+            return this.collator.compare(s1.sortName, s2.sortName);
```

Comment:
Can produce an NPE either use a lambda to filter nulls or the NullSafeComparator i.e.
```return new NullSafeComparator<>(collator, false).compare(s1.getSortName(), s2.getSortName()```

Reply:
Done!